frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx

```diff
@@ -0,0 +1,257 @@
+import { useLeaseDepositReturns } from './hooks/useDepositReturns';
+import { useLeaseDeposits } from './hooks/useDeposits';
+import ReceivedDepositModal from './modal/receivedDepositModal/ReceivedDepositModal';
+import ReturnedDepositModal from './modal/returnedDepositModal/ReturnedDepositModal';
+import * as Styled from './styles';
+
+export interface IDepositsContainerProps {}
+
+export const DepositsContainer: React.FunctionComponent<IDepositsContainerProps> = () => {
+  const { lease, setLease } = useContext(LeaseStateContext);
+  const { values } = useFormikContext<IFormLease>();
+  const securityDeposits: ILeaseSecurityDeposit[] = getIn(values, 'securityDeposits') ?? [];
+  const depositReturns: ILeaseSecurityDepositReturn[] =
+    getIn(values, 'securityDepositReturns') ?? [];
+
+  const [showDepositEditModal, setShowEditModal] = useState<boolean>(false);
+  const [deleteModalWarning, setDeleteModalWarning] = useState<boolean>(false);
+  const [deleteReturnModalWarning, setDeleteReturnModalWarning] = useState<boolean>(false);
+
+  const [showReturnEditModal, setShowReturnEditModal] = useState<boolean>(false);
+
+  const [depositToDelete, setDepositToDelete] = useState<FormLeaseDeposit | undefined>(undefined);
+  const [editDepositValue, setEditDepositValue] = useState<FormLeaseDeposit>(
+    FormLeaseDeposit.createEmpty(),
+  );
+
+  const [depositReturnToDelete, setDepositReturnToDelete] = useState<
+    FormLeaseDepositReturn | undefined
+  >(undefined);
+  const [editReturnValue, setEditReturnValue] = useState<FormLeaseDepositReturn | undefined>(
+    undefined,
+  );
+
+  const { updateLeaseDeposit, removeLeaseDeposit } = useLeaseDeposits();
+  const { updateLeaseDepositReturn, removeLeaseDepositReturn } = useLeaseDepositReturns();
+
+  const onAddDeposit = () => {
+    setEditDepositValue(FormLeaseDeposit.createEmpty());
+    setShowEditModal(true);
+  };
+
+  const onEditDeposit = (id: number) => {
+    var deposit = securityDeposits.find(x => x.id === id);
+    if (deposit) {
+      setEditDepositValue(FormLeaseDeposit.createFromModel(deposit));
+      setShowEditModal(true);
+    }
+  };
+
+  const onDeleteDeposit = (id: number) => {
+    var deposit = securityDeposits.find(x => x.id === id);
+    if (deposit) {
+      setDepositToDelete(FormLeaseDeposit.createFromModel(deposit));
+      setDeleteModalWarning(true);
+    }
+  };
+
+  const onReturnDeposit = (id: number) => {
+    var deposit = securityDeposits.find(x => x.id === id);
+    if (deposit) {
+      setEditReturnValue(FormLeaseDepositReturn.createEmpty(deposit));
+      setShowReturnEditModal(true);
+    }
+  };
+
+  const onDeleteDepositConfirmed = async () => {
+    if (lease && lease.id && lease.rowVersion && depositToDelete) {
+      const updatedLease = await removeLeaseDeposit({
+        parentId: lease.id,
+        parentRowVersion: lease.rowVersion,
+        payload: FormLeaseDeposit.toInterfaceModel(depositToDelete),
```

Comment:
I'm kind of surprised you went with a static method here, wouldn't operating on the data in the instance make sense?

Reply:
thats right, I will change it to operate on the object directly